include/crm/common/xml_compat.h

```diff
@@ -212,6 +212,9 @@ xmlNode *expand_idref(xmlNode *input, xmlNode *top);
 //! \deprecated Do not use Pacemaker for general-purpose XML manipulation
 void crm_xml_set_id(xmlNode *xml, const char *format, ...) G_GNUC_PRINTF(2, 3);
 
+//! \deprecated Do not use Pacemaker for general-purpose XML manipulation
+void crm_xml_sanitize_id(char *id);
```

Comment:
It occurs to me we need to keep track of new deprecations (since 2.1.8-rc1) since we don't want to remove those in 3.0.0. Alternatively drop the old deprecations first

Reply:
I hate to postpone all these commits further but dropping old deprecations first is likely the better approach.

Reply:
done as of several PRs today